python/ray/llm/_internal/serve/deployments/llm/vllm/vllm_engine.py

```
@@ -305,12 +305,49 @@ async def start(self):
         logger.info("Started vLLM engine.")
 
     async def _start_engine(self) -> "EngineClient":
+        from vllm import envs
+
+        # Since vLLM 0.8.0, the logic to determine v0/v1 engine is as follows:
+        # 1. If VLLM_USE_V1 is not set, then it tries to use v1 engine. However,
+        #    if any feature specified in the engine config is not supported, then
+        #    it falls back to v0. Note that launching vLLM on a non-main thread
+        #    is an experimental feature, so vLLM will fall back to v0 in this case.
+        # 2. If VLLM_USE_V1 is set to 1, then it will use v1 engine even with
+        #    experimental features (such as launching vLLM on a non-main thread).
+        # 3. If VLLM_USE_V1 is set to 0, force using v0 engine.
+        # In Ray Serve LLM, we forbid case 1 because we have to know exactly which engine is used.
+        if not envs.is_set("VLLM_USE_V1"):
+            raise AssertionError(
+                "Starting from Ray 2.45, VLLM_USE_V1 environment variable must be "
```

Should we just default to vllm v0? If we really want to raise this assertion error, I think we should update the doc and gen config as well

---

Discussed with Kourosh, will default to v0 for now. And drop this PR in favor of https://github.com/ray-project/ray/pull/51726